integrations/extensions/starter-kits/search-with-milvus/search-with-watsonx-data-milvus.md

```diff
@@ -0,0 +1,117 @@
+2. On the left-side panel, select `watsonx.data Milvus` as the vector store --> fill in name and description --> select the Milvus connection created earlier. 
+3. Select the Milvus connection created earlier -> Select `Database` and `Embeddings model` from the dropdowns --> click `Next`. For example,  
+<img src="./assets/create-milvus-index-watsonx-ai.png" width="1080" height="574" />
+
+4. Click `New collection` to create a new collection.
+5. Provide a unique collection name and choose files to add to the Milvus collection --> click `Create`.
+6. Once the document uploading process is done, you can start testing it in the prompt lab.
+
+The following screen recording is a demonstration of the above steps: [./assets/create-milvus-index-watsonx-ai.mov](./assets/create-milvus-index-watsonx-ai.mov).
+
+**NOTE: `document_name` and `text` are the two main fields created in the Milvus collection schema by default. When searching this Milvus collection using custom code, you need to specify these two fields as output_fields. When setting up the built-in Milvus search integration, you need to configure the Title and Body fields with these two fields.**
+
+### Option 2: Ingest data using custom code
+Here is a sample code to ingest documents into Milvus: [../search-with-custom-service/examples/index-with-milvus.py](../search-with-custom-service/examples/index-with-milvus.py). To run it, 
+1. Install dependencies:
+   ```bash
+   python3 -m pip install pymilvus langchain langchain-milvus langchain-ibm ibm-watsonx-ai PyPDF2
+   ```
+2. Create environment variables for Milvus credentials
+  ```bash
+  export MILVUS_HOST="Your Milvus GRPC host"
+  export MILVUS_PORT="Your Milvus GRPC port"
+  export MILVUS_USER="ibmlhapikey" // The default username for watsonx.data Milvus
+  export MILVUS_PASSWORD="Your watsonx.data API key"
+
+  export MILVUS_COLLECTION_NAME="Your Milvus collection name" // It can be anything
+  export WATSONX_AI_APIKEY="Your watsonx.ai API key" // watsonx.ai embeddings model is used to create vectors
+  export WATSONX_AI_PROJECT_ID="Your watsonx.ai project ID" // watsonx.ai project ID is required to access the embeddings models
+  ```
+3. Update the `SOURCE_FILES`, `SOURCE_URLS`, and `SOURCE_TITLES` variables at the begining of the script to your file names, urls, and titles respectively.
+4. Run the script
+   ```bash
+   python3 index-with-milvus.py
+   ```
+
+## Step 3: Connect to watsonx Orchestrate or Assistant
+There are two ways to connect your Milvus vector database to watsonx Orchestrate or Assistant for search and conversational search, either through the built-in Milvus search integration or using the Custom Service search.
+
+**NOTE: The embeddings model(s) you use for search using either option in [Step 3](#step-3-connect-to-watsonx-orchestrate-or-assistant) has to align with the embeddings model(s) used for data ingestion in [Step 2](#step-2-ingest-data-into-milvus).**
```

Comment:
Small point: Why say "embeddings model(s)" rather than just "embeddings model?"
The full sentence could be:


NOTE: The embeddings model you use for search using either option in [Step 3](https://github.com/watson-developer-cloud/assistant-toolkit/blob/303f5082ce967264c42d145c83a610bf6e9bc401/integrations/extensions/starter-kits/search-with-milvus/search-with-watsonx-data-milvus.md#step-3-connect-to-watsonx-orchestrate-or-assistant) must align with the embeddings model used for data ingestion in [Step 2](https://github.com/watson-developer-cloud/assistant-toolkit/blob/303f5082ce967264c42d145c83a610bf6e9bc401/integrations/extensions/starter-kits/search-with-milvus/search-with-watsonx-data-milvus.md#step-2-ingest-data-into-milvus).